pkg/policy/api/rule_validation.go

```diff
@@ -259,12 +258,32 @@ func (pr *L7Rules) sanitize() error {
 	}
 
 	if pr.Kafka != nil {
+		nTypes++
 		for i := range pr.Kafka {
 			if err := pr.Kafka[i].Sanitize(); err != nil {
 				return err
 			}
 		}
 	}
+
+	if pr.L7 != nil && pr.L7Proto == "" {
+		return fmt.Errorf("'l7' may only be specified when a 'l7proto' is also specified")
+	}
+	if pr.L7Proto != "" {
+		if pr.L7 == nil {
+			return fmt.Errorf("'l7' must be specified when 'l7proto' is specified")
```

Comment:
Non-blocking. Do we really need to enforce this? We may want to provide parsers which do not require configuration at all.

Reply:
Removed the nil check as range on nil slice works as intended.